SRC/cgedmd.f90

```diff
@@ -0,0 +1,995 @@
+         MWRKEV = MAX( 1, 2*N )
+         MLWORK = MAX(MLWORK,MWRKEV) 
+         MLRWRK = MAX(MLRWRK,N+2*N)
+         IF ( LQUERY ) THEN 
+             CALL CGEEV( 'N', JOBZL, N, S, LDS, EIGS, & 
+              W, LDW, W, LDW, ZWORK, -1, RWORK, INFO1 ) ! LAPACK CALL
+                LWRKEV = INT(ZWORK(1))
+                OLWORK = MAX( OLWORK, LWRKEV )
+                OLWORK = MAX( 2, OLWORK )
+         END IF 
+!
+         IF ( LIWORK < IMINWR .AND. (.NOT.LQUERY) ) INFO = -30
+         IF ( LRWORK < MLRWRK .AND. (.NOT.LQUERY) ) INFO = -28
+         IF ( LZWORK < MLWORK .AND. (.NOT.LQUERY) ) INFO = -26
+         
+      END IF
+!      
+      IF( INFO /= 0 ) THEN
+         CALL XERBLA( 'CGEDMD', -INFO )
+         RETURN
+      ELSE IF ( LQUERY ) THEN
+!     Return minimal and optimal workspace sizes
+          IWORK(1) = IMINWR
+          RWORK(1) = MLRWRK
+          ZWORK(1) = MLWORK 
+          ZWORK(2) = OLWORK
+          RETURN
+      END IF   
+!............................................................
+!      
+      OFL   = SLAMCH('O')*SLAMCH('P')
+      SMALL = SLAMCH('S')
+      BADXY = .FALSE.
+!
+!     <1> Optional scaling of the snapshots (columns of X, Y)
+!     ==========================================================
+      IF ( SCCOLX ) THEN 
+          ! The columns of X will be normalized. 
+          ! To prevent overflows, the column norms of X are 
+          ! carefully computed using CLASSQ.    
+          K = 0 
+          DO i = 1, N
+            !WORK(i) = SCNRM2( M, X(1,i), 1 )  
+            SCALE  = ZERO
+            CALL CLASSQ( M, X(1,i), 1, SCALE, SSUM )
+            IF ( SISNAN(SCALE) .OR. SISNAN(SSUM) ) THEN
+                K    =  0 
+                INFO = -8 
+                CALL XERBLA('CGEDMD',-INFO)
+            END IF 
+            IF ( (SCALE /= ZERO) .AND. (SSUM /= ZERO) ) THEN 
+               ROOTSC = SQRT(SSUM)
+               IF ( SCALE .GE. (OFL / ROOTSC) ) THEN 
+!                 Norm of X(:,i) overflows. First, X(:,i) 
+!                 is scaled by
+!                 ( ONE / ROOTSC ) / SCALE = 1/||X(:,i)||_2. 
+!                 Next, the norm of X(:,i) is stored without
+!                 overflow as WORK(i) = - SCALE * (ROOTSC/M), 
+!                 the minus sign indicating the 1/M factor.  
+!                 Scaling is performed without overflow, and
+!                 underflow may occur in the smallest entries
+!                 of X(:,i). The relative backward and forward
+!                 errors are small in the ell_2 norm.                      
+                  CALL CLASCL( 'G', 0, 0, SCALE, ONE/ROOTSC, &
+                               M, 1, X(1,i), LDX, INFO1 )                 
+                  RWORK(i) = - SCALE * ( ROOTSC / FLOAT(M) )
+               ELSE 
+!                 X(:,i) will be scaled to unit 2-norm
+                  RWORK(i) =   SCALE * ROOTSC 
+                  CALL CLASCL( 'G',0, 0, RWORK(i), ONE, M, 1, &

```

Comment:
Is there a particular reason to use CLASCL and not CSSCAL?

Reply:
No, not to my knowledge. I will reach out to Zlatko regarding this as it was his choice, but I assume no.

Reply:
Now thinking about this, I believe CLASCL was used so `RWORK(i)` could be used implicitly compute `( 1 / RWORK(i) )` without creating an extra variable to do so.

If there is a preference to having a call to CSSCAL instead, I can make this change and propagate such changes elsewhere.